src/server/pps/server/api_server.go

```diff
@@ -651,6 +651,45 @@ func (a *apiServer) ListJobSet(request *pps.ListJobSetRequest, serv pps.API_List
 
 	// Track the jobsets we've already processed
 	seen := map[string]struct{}{}
+	// Prepare filter functions.
+	projectsFilter := make(map[string]bool, len(request.GetProjects()))
+	for _, project := range request.GetProjects() {
+		projectsFilter[project.GetName()] = true
+	}
+	jqFilter, err := jqFilterFunc(serv.Context(), request.GetJqFilter())
```

Comment:
The new variable `jqFilter` here isn’t really a _jq_ filter; it’s just a message filter, specifically a job info filter. Maybe call it `jobInfoFilter` or just `filter`?